dev/update_changelog.py

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+"""Script for automatically updating the `CHANGELOG.md` based on the commits since the latest release tag."""
+from pathlib import Path
+import re
+import subprocess
+
+DEFAULT_CHANGELOG_SECTIONS = """
+### ‼️ Breaking changes
+
+
+### ✨ New features
+
+
+### 🗑️ Deprecation
+
+
+### 👌 Improvements
+
+
+### 🐛 Bug Fixes
+
+
+### 📚 Documentation
+
+
+### 🔧 Maintenance
+
+
+### ⬆️ Update dependencies
+
+
+### ♻️ Refactor
+
+"""
+
+
+def update_changelog():
+    """Update the `CHANGELOG.md` for a first draft of the release."""
+
+    print('🔍 Checking the current version number')
+    with Path('CHANGELOG.md').open('r', encoding='utf8') as handle:
+        current_changelog = handle.read()
+
+    with Path('src/aiida_quantumespresso/__init__.py').open('r', encoding='utf8') as handle:
+        version = re.search(r"__version__ = '(?P<version_number>\d+.+)'", handle.read()).groupdict()['version_number']
+
+    if str(version) in current_changelog:
```

Comment:
This is a bit fragile I guess. It is possible that a changelog will contain a future version in its notes. For example, it could include "Deprecated `PwBaseWorkChain`, will be removed in v15.0.0`. Guess it might be unlikely but maybe can be easily made more robust by checking that the version appears in a line starting with `##` as in a markdown section header?

Reply:
I think now we can also just really grab the version from the installed `aiida_quantumespresso` instance and convert it into a string? I didn't do that here because I wanted the check on GitHub to be fast and so I wanted to avoid installing the package.